public/js/reducers/sources.js

```diff
@@ -16,7 +16,9 @@ const initialState = fromJS({
 function update(state = initialState, action) {
   switch (action.type) {
     case constants.ADD_SOURCE:
-      return state.mergeIn(["sources", action.source.actor], action.source);
+      let newSource = action.source;
```

Comment:
The only downside to adding more "derived" properties in the state is that any of our mocks will have to also specify it. I know I said that you can "normalize" data here but to me this is more "derived data". But I don't really care that much, if you think it should be here that's fine!